packages/database/src/TupaiaDatabase.js

```diff
@@ -683,26 +682,13 @@ function addJoin(baseQuery, recordType, joinOptions) {
 }
 
 function getColSelector(connection, inputColStr) {
-  const regexp = new RegExp(/->>?/, 'g');
-  if (regexp.test(inputColStr)) {
-    // Shorthand way of querying json property
-    // TODO: Replace with knex json where functions, eg. whereJsonPath
-    const [first, ...rest] = inputColStr.split(regexp);
-    if (rest.length === 1) {
-      // get the first separator (there will only be 1 in this case)
-      const separator = inputColStr.match(regexp)[0];
-      // e.g. 'config->>colour' is converted to config->>'colour' and 'config->colour' is converted to config->'colour'
-      return connection.raw(`??${separator}?`, [first, ...rest]);
-    }
-    // e.g. 'config->item->>colour' is converted to config->'item'->>'colour'
-    const last = rest.slice(-1);
-    const middle = rest.slice(0, rest.length - 1);
-    return connection.raw(`??->${middle.map(i => '?').join('->')}->>?`, [
-      first,
-      ...middle,
-      ...last,
-    ]);
-  }
+  const jsonOperatorPattern = /->>?/g;
+  if (!jsonOperatorPattern.test(inputColStr)) return inputColStr;
+
+  const params = inputColStr.split(jsonOperatorPattern);
+  const allButFirst = params.slice(1);
 
-  return inputColStr;
+  // Turn `config->item->>colour` into `config #>> '{item,colour}'`
+  // For some reason, Knex fails when we try to convert it to `config->'item'->>'colour'`
+  return connection.raw(`?? #>> '{${allButFirst.map(() => '?').join(',')}}'`, params);
```

Comment:
@rohan-bes Slightly deviated from your suggestion to avoid performance cost of `...` rest operator

Reply:
Ah nice call!